executor/cte_table_reader.go

```diff
@@ -0,0 +1,78 @@
+// Copyright 2021 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package executor
+
+import (
+	"context"
+
+	"github.com/pingcap/errors"
+	"github.com/pingcap/tidb/util/chunk"
+	"github.com/pingcap/tidb/util/cteutil"
+)
+
+// CTETableReaderExec scans data in iterInTbl, which is filled by corresponding CTEExec.
+type CTETableReaderExec struct {
+	baseExecutor
+
+	iterInTbl cteutil.Storage
+	chkIdx    int
+	curIter   int
```

Comment:
`curIter` is useless

Reply:
Related to this [comment](https://github.com/pingcap/tidb/pull/24809/files#r641432837)